data/Templates/eCR/Header.liquid

```diff
@@ -1,8 +1,14 @@
 {% evaluate practitionerId using 'Utils/GenerateId' obj: msg.ClinicalDocument.componentOf.encompassingEncounter.responsibleParty.assignedEntity -%}
-{% if msg.ClinicalDocument.id and msg.ClinicalDocument.id.root -%}
-  {% assign compositionId = msg.ClinicalDocument.id.root -%}
+{% if msg.ClinicalDocument.id -%}
+  {% if msg.ClinicalDocument.id.extension -%}
+    {% comment %} If extension exists, use that {% endcomment %}
+    {% assign compositionId = msg.ClinicalDocument.id.extension -%}
+  {% else if msg.ClinicalDocument.id.root -%}
+    {% comment %} If extension does not exist and root does, use that {% endcomment %}
+    {% assign compositionId = msg.ClinicalDocument.id.root -%}
+  {% endif -%}
 {% else -%}
-  {% assign compositionId = msg._originalData | generate_uuid -%}
+  {% assign compositionId =msg._originalData | generate_uuid -%}
```

Comment:
is it possible to have an id, but not an extension or root such that `compositionID` would end up undefined here?  (maybe to be safe we pre-assign this version, then do an else-less if on the id branch?